### PR TITLE
fix: Added 50k overhead to userOps estimation

### PR DESCRIPTION
**Pull request by farhanW3**
## PR-Codex overview
This PR focuses on improving gas estimations, updating contract addresses, and enhancing signature verification in the codebase.

### Detailed summary
- Added 50k overhead to userOp gas estimations
- Updated MANAGED_ACCOUNT_GAS_BUFFER constant
- Improved bytecode caching mechanism
- Updated paymaster URL format
- Enhanced signature verification for smart contract wallets
- Updated data types in EstimateUserOpGasResult
- Updated contract addresses from USDC to USDT
- Improved simulation and gas estimation functions for ERC20 transactions

> The following files were skipped due to too many changes: `packages/thirdweb/src/wallets/smart/lib/bundler.ts`, `packages/thirdweb/src/transaction/actions/estimate-gas-cost.test.ts`, `packages/thirdweb/src/extensions/erc721/write/sigMint.test.ts`, `packages/thirdweb/src/transaction/actions/encode.test.ts`, `packages/thirdweb/src/event/actions/get-events.test.ts`, `packages/thirdweb/src/transaction/actions/encode.bench.ts`, `legacy_packages/wallets/src/evm/connectors/smart-wallet/lib/base-api.ts`, `legacy_packages/wallets/src/evm/connectors/smart-wallet/lib/http-rpc-client.ts`, `packages/thirdweb/src/transaction/actions/send-transaction.test.ts`, `packages/thirdweb/src/wallets/smart/index.ts`, `packages/thirdweb/src/transaction/actions/send-and-confirm-transaction.test.ts`, `packages/thirdweb/src/wallets/smart/smart-wallet-integration.test.ts`, `packages/thirdweb/test/src/abis/usdt.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: legacy_packages/wallets/src/evm/connectors/smart-wallet/lib/base-api.ts [hunk 1]
```diff
@@ -333,6 +333,9 @@ export abstract class BaseAccountAPI {
       );
     }
 
+    // add 50k overhead for entrypoint checks (following same as unity-sdk)
+    partialOp.callGasLimit = partialOp.callGasLimit.add(50_000);
```
Comment (joaquim-verges) [inline legacy_packages/wallets/src/evm/connectors/smart-wallet/lib/base-api.ts]: you need to do this in diff places depending on the path here.

line 284
line 299
line 327

File: packages/thirdweb/src/wallets/smart/lib/userop.ts [hunk 2]
```diff
@@ -147,6 +147,9 @@ export async function createUnsignedUserOp(args: {
     partialOp.preVerificationGas = estimates.preVerificationGas;
   }
 
+  // add 50k overhead for entrypoint checks (following same as unity-sdk)
+  partialOp.callGasLimit = partialOp.callGasLimit + 50_000n;
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/smart/lib/userop.ts]: same here - there's diff path including one that needs to re-sign the payload from the paymaster

**Review by joaquim-verges — CHANGES_REQUESTED** [review]
test caught an issue here

**Review by 0xFirekeeper — APPROVED** [review]

File: legacy_packages/wallets/src/evm/connectors/smart-wallet/lib/base-api.ts [hunk 3]
```diff
@@ -296,7 +296,10 @@ export abstract class BaseAccountAPI {
         } catch (error: any) {
           throw this.unwrapBundlerError(error);
         }
-        partialOp.callGasLimit = BigNumber.from(estimates.callGasLimit);
+        // add 50k overhead for entrypoint checks (following same as unity-sdk)
+        partialOp.callGasLimit = BigNumber.from(estimates.callGasLimit).add(
```
Comment (joaquim-verges) [inline legacy_packages/wallets/src/evm/connectors/smart-wallet/lib/base-api.ts:300]: missing one case line 284

File: packages/thirdweb/src/wallets/smart/lib/userop.ts [hunk 4]
```diff
@@ -119,7 +119,8 @@ export async function createUnsignedUserOp(args: {
         userOp: partialOp,
         options,
       });
-      partialOp.callGasLimit = estimates.callGasLimit;
+      // add 50k overhead for entrypoint checks (following same as unity-sdk)
+      partialOp.callGasLimit = estimates.callGasLimit + 50_000n;
```
Comment (joaquim-verges) [inline packages/thirdweb/src/wallets/smart/lib/userop.ts:123]: missing one case line 113.

also can you remove `(following same as unity-sdk)` please

**Review by joaquim-verges — CHANGES_REQUESTED** [review]

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**